ArcGISRuntimeSDKQt_CppSamples/Maps/OpenMobileMap_MapPackage/OpenMobileMap_MapPackage.cpp

```diff
@@ -38,30 +111,67 @@ void OpenMobileMap_MapPackage::init()
 
 void OpenMobileMap_MapPackage::componentComplete()
 {
-    QQuickItem::componentComplete();
+  QQuickItem::componentComplete();
 
-    // find QML MapView component
-    m_mapView = findChild<MapQuickView*>("mapView");
+  // find QML MapView component
+  m_mapView = findChild<MapQuickView*>("mapView");
 
-    // get the data path
-    QString dataPath = QQmlProperty::read(this, "dataPath").toUrl().toLocalFile();
+  // get the data path
+  const QString dataPath = defaultDataPath() + sampleFileYellowstone;
 
-    //! [open mobile map package cpp snippet]
-    // instatiate a mobile map package
-    m_mobileMapPackage = new MobileMapPackage(dataPath + "Yellowstone.mmpk", this);
+  // Check if the MMPK can be read directly
+  MobileMapPackage::isDirectReadSupported(dataPath);
+}
 
-    // wait for the mobile map package to load
-    connect(m_mobileMapPackage, &MobileMapPackage::doneLoading, this, [this](Error error)
+// Slot for handling when the package loads
+void OpenMobileMap_MapPackage::packageLoaded(const Error& e)
+{
+  if (!e.isEmpty())
+  {
+    qDebug() << QString("Package load error: %1 %2").arg(e.message(), e.additionalMessage());
+    return;
+  }
+
+  setMapToView();
+}
+
+// create map package and connect to signals
+void OpenMobileMap_MapPackage::createMapPackage(const QString& path)
+{
+  // instatiate a mobile map package
+  m_mobileMapPackage = new MobileMapPackage(path, this);
+
+  // wait for the mobile map package to load
+  connect(m_mobileMapPackage, &MobileMapPackage::doneLoading, this, [this](Error error)
+  {
+    if (error.isEmpty())
     {
-        if (error.isEmpty())
-        {
-            // set the map view's map to the first map in the mobile map package
-            m_mapView->setMap(m_mobileMapPackage->maps().at(0));
-        }
-    });
-
-    // load the mobile map package
-    m_mobileMapPackage->load();
-    //! [open mobile map package cpp snippet]
+      // set the map view's map to the first map in the mobile map package
+      setMapToView();
+    }
+  });
+
+  connect(m_mobileMapPackage, &MobileMapPackage::doneLoading, this, &OpenMobileMap_MapPackage::packageLoaded);
+
```

Comment:
I'd move the empty line after the closing tag of the snippet

Reply:
Done.